File: ios/Gomoku/GameMode.swift
import Foundation

enum GameMode: CaseIterable {
    case pvp
    case pveHard
    case pveMedium
    case pveEasy

    var aiDepth: Int {
        switch self {
        case .pvp: return 0
        case .pveHard: return 4
        case .pveMedium: return 2
        case .pveEasy: return 1
        }
    }

    var candidateRange: Int {
        switch self {
        case .pvp: return 0
        case .pveHard: return 1
        case .pveMedium: return 2
        case .pveEasy: return 3
        }
    }

    var maxCandidates: Int {
        switch self {
        case .pvp: return 0
        case .pveHard: return 10
        case .pveMedium: return 12
        case .pveEasy: return 15
        }
    }

    var isPVE: Bool {
        return self != .pvp
    }

    var displayName: String {
        switch self {
        case .pvp: return "人人对战"
        case .pveHard: return "AI 高级"
        case .pveMedium: return "AI 中级"
        case .pveEasy: return "AI 普通"
        }
    }
}

File: android/app/src/main/java/org/example/gomoku/GameMode.java
package org.example.gomoku;

/**
 * 游戏模式
 */
public enum GameMode {
    /** 人人对战 */
    PVP(0, 0, 0),
    /** 高级人机 */
    PVE_HARD(4, 2, 20),
    /** 中级人机 */
    PVE_MEDIUM(3, 2, 30),
    /** 普通人机 */
    PVE_EASY(1, 1, 0);

    private final int aiDepth;
    private final int candidateRange;
    private final int maxCandidates;

    GameMode(int aiDepth, int candidateRange, int maxCandidates) {
        this.aiDepth = aiDepth;
        this.candidateRange = candidateRange;
        this.maxCandidates = maxCandidates;
    }

    public int getAiDepth() {
        return aiDepth;
    }

    public int getCandidateRange() {
        return candidateRange;
    }

    public int getMaxCandidates() {
        return maxCandidates;
    }

    public boolean isPVE() {
        return this != PVP;
    }
}
